mantle/util/common.go

```diff
@@ -120,3 +121,43 @@ func RunCmdTimeout(timeout time.Duration, cmd string, args ...string) error {
 		return fmt.Errorf("%s timed out after %s", cmd, timeout)
 	}
 }
+
+// ParseDiskSpec converts a disk specification into a Disk. The format is:
+// <size>[:<opt1>,<opt2>,...], like ["5G:channel=nvme"]
+func ParseDiskSpec(spec string) (string, map[string]string, error) {
+	diskmap := map[string]string{}
+	split := strings.Split(spec, ":")
+	if split[0] == "" {
+		return "", nil, fmt.Errorf("invalid size %s", spec)
+	}
+	var disksize string
+	if len(split) == 1 {
+		disksize = split[0]
+	} else if len(split) == 2 {
+		disksize = split[0]
+		for _, opt := range strings.Split(split[1], ",") {
+			if opt == "mpath" {
+				diskmap["MultiPathDisk"] = "true"
+			} else if opt == "4k" {
+				diskmap["SectorSize"] = "4096"
+			} else if strings.HasPrefix(opt, "channel=") {
+				channel := strings.TrimPrefix(opt, "channel=")
+				if channel == "" {
+					return "", nil, fmt.Errorf("invalid channel opt %s", opt)
+				}
+				diskmap["Channel"] = channel
+			} else if strings.HasPrefix(opt, "serial=") {
+				serial := strings.TrimPrefix(opt, "serial=")
+				if serial == "" {
+					return "", nil, fmt.Errorf("invalid serial opt %s", opt)
+				}
+				diskmap["DeviceOpts"] = "serial=" + serial
+			} else {
+				return "", nil, fmt.Errorf("unknown disk option %s", opt)
+			}
```

Comment:
I was thinking something more generic here where it just parses any key/value and stores it in the map (for things like `mpath` and `4k` that don't have values, the value can just be the empty string).

So this parser wouldn't know anything about multipath, sector sizes, etc...

Reply:
What about if `opt == "4k"`, then `diskmap["SectorSize"] = "4096"`, does this make sense?

Reply:
The idea is that this function doesn't have any knowledge of knobs specific to a platform. It would just be about parsing the disk spec format into something more easily consumable.

Then, e.g. the QEMU platform can check if `4k` was provided and it knows that it means setting the sector size to 4096.